### PR TITLE
refactor: unify stop and teardown to shutdown

### PR DESCRIPTION
**Pull request by JoanFM**
**Goals:**
Try to simplify Gateway API by having one less interace to implement

### CODE REVIEW

File: jina/serve/gateway.py [hunk 1]
```diff
@@ -143,13 +143,9 @@ async def run_server(self):
...
 
-    async def teardown(self):
-        """Free other resources allocated with the server, e.g, gateway object, ..."""
-        await self.streamer.close()
-
     @abc.abstractmethod
-    async def stop_server(self):
-        """Stop server"""
+    async def shutdown(self):
+        """Free other resources allocated with the server, e.g, gateway object, ..."""
```
Comment (alaeddine-13) [inline jina/serve/gateway.py:148]:
```suggestion
        """Shutdown the server and free other allocated resources, e.g, streamer object, health check service, ..."""
```

File: tests/docker_compose/custom-gateway/dummy_gateway.py [hunk 2]
```diff
@@ -69,13 +69,7 @@ async def _process(text: str):
...
         await self.server.serve()
 
-    async def teardown(self):
-        await super().teardown()
-        await self.server.shutdown()
-
-    async def stop_server(self):
+    async def shutdown(self):
+        await self.streamer.close()
         self.server.should_exit = True
```
Comment (alaeddine-13) [inline tests/docker_compose/custom-gateway/dummy_gateway.py:74]: we don't need it anymore right ?

File: jina/serve/runtimes/gateway/grpc/gateway.py [hunk 3]
```diff
@@ -96,16 +96,11 @@ async def setup_server(self):
...
+    async def shutdown(self):
         """Free other resources allocated with the server, e.g, gateway object, ..."""
-        await super().teardown()
-        await self.health_servicer.enter_graceful_shutdown()
-
-    async def stop_server(self):
-        """
-        Stop GRPC server
-        """
+        await self.streamer.close()
```
Comment (alaeddine-13) [inline jina/serve/runtimes/gateway/grpc/gateway.py:101]: can't we put this repetitive line in base class so users don't have to worry about the streamer object ?

File: tests/docker_compose/custom-gateway/dummy_gateway.py (hunk 2 above)
Comment (JoanFM) [reply tests/docker_compose/custom-gateway/dummy_gateway.py:74]: we do

File: jina/serve/runtimes/gateway/grpc/gateway.py (hunk 3 above)
Comment (JoanFM) [reply jina/serve/runtimes/gateway/grpc/gateway.py:101]: Okey, I will have `GatewayRuntime` close it

Comment (JoanFM) [reply jina/serve/runtimes/gateway/grpc/gateway.py]: Done

**Pull request merged**